include/aikido/control/BarrettFingerKinematicSimulationSpreadCommandExecutor.hpp

```diff
@@ -0,0 +1,96 @@
+#ifndef AIKIDO_CONTROL_BARRETFINGERKINEMATICSIMULATIONSPREADCOMMANDEXECUTOR_HPP_
+#define AIKIDO_CONTROL_BARRETFINGERKINEMATICSIMULATIONSPREADCOMMANDEXECUTOR_HPP_
+#include <dart/collision/CollisionDetector.hpp>
+#include <dart/collision/CollisionOption.hpp>
+#include <dart/collision/CollisionGroup.hpp>
+#include <dart/collision/CollisionFilter.hpp>
+#include <dart/dynamics/dynamics.hpp>
+#include <future>
+#include <mutex>
+#include <condition_variable>
+#include <chrono>
+
+namespace aikido {
+namespace control {
+
+/// This executor mimics the spread behavior of BarretFinger.
+/// It moves two finger spreads simultaneously to certain goal value;
+/// it will stop prematurely if joint limit is reached or collision is detected.
+class BarrettFingerKinematicSimulationSpreadCommandExecutor
+{
+public:
+  /// Constructor.
+  /// \param[in] fingers 2 fingers to be controlled by this Executor.
+  /// \param[in] spread Index of spread dof
+  /// \param[in] collideWith CollisionGroup to check collision with fingers.
+  /// \param[in] collisionOptions Default is (enableContact=false, binaryCheck=true,
+  ///        maxNumContacts = 1.)
+  BarrettFingerKinematicSimulationSpreadCommandExecutor(
+    std::array<::dart::dynamics::ChainPtr, 2> fingers, size_t spread,
+    ::dart::collision::CollisionGroupPtr collideWith,
+    ::dart::collision::CollisionOption collisionOptions
+      = ::dart::collision::CollisionOption(false, 1));
+
+  /// Sets variables to move the spread joint by goalPosition,
+  /// joint limit has reached, or until collision is detected.
+  /// Must call step function after this for actual execution.
+  /// \param goalPosition Desired angle of spread joint.
+  /// \return future which becomes available when the execution completes.
+  std::future<void> execute(double goalPosition);
+
+  /// Moves the joint of the finger by fixed speed*timeSincePreviousCall
+  /// until execute's goalPosition by spread dof or collision is detected.
+  /// If multiple threads are accessing this function or the skeleton associated
+  /// with this executor, it is necessary to lock the skeleton before
+  /// calling this method.
+  /// \param[in] timeSincePreviousCall Time interval to take.
+  void step(const std::chrono::milliseconds& timeSincePreviousCall);
```

Comment:
This call would always be called by `BarrettHandKinematicSimulationPositionCommandExecutor::step()` method, so I am not sure whether a default is necessary. We can set it to something I guess... :)